assets/app/scripts/controllers/deployment.js

```diff
@@ -0,0 +1,182 @@
+'use strict';
+
+/**
+ * @ngdoc function
+ * @name openshiftConsole.controller:PodController
+ * @description
+ * # ProjectController
+ * Controller of the openshiftConsole
+ */
+angular.module('openshiftConsole')
+  .controller('DeploymentController', function ($scope, $routeParams, DataService, project, DeploymentsService, ImageStreamResolver, $filter) {
+    $scope.deployment = null;
+    $scope.deploymentConfig = null;
+    $scope.deployments = {};
+    $scope.podTemplates = {};
+    $scope.imageStreams = {};
+    $scope.imagesByDockerReference = {};
+    $scope.imageStreamImageRefByDockerReference = {}; // lets us determine if a particular container's docker image reference belongs to an imageStream
+    $scope.builds = {};     
```

Comment:
Is all of this needed for the pod template? It seems like a lot to watch for the amount of data we show.